undertow/src/main/java/org/wildfly/extension/undertow/Server.java

```diff
@@ -162,6 +163,9 @@ public List<UndertowListener> getListeners() {
     }
 
     public String getRoute() {
+        if (this.finalRoute != null) {
```

Comment:
Since this is a dynamically generated value, it would be best to establish this within Service.start(...).

Reply:
Note that doing that may require changes to some unit test stuff that uses a Server but doesn't call start/stop on it.

org.wildfly.mod_cluster.undertow.UndertowEngineTestCase, perhaps others.